parallel-consumer-reactor/src/test/java/io/confluent/parallelconsumer/reactor/ReactorPCTest.java

```diff
@@ -0,0 +1,82 @@
+package io.confluent.parallelconsumer.reactor;
+
+import io.confluent.csid.utils.StringUtils;
+import io.confluent.parallelconsumer.ParallelConsumerOptions;
+import io.confluent.parallelconsumer.ParallelEoSStreamProcessor;
+import io.confluent.parallelconsumer.ParallelEoSStreamProcessorTestBase;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.stream.BaseStream;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+import static io.confluent.parallelconsumer.ParallelConsumerOptions.CommitMode.PERIODIC_CONSUMER_SYNC;
+import static io.confluent.parallelconsumer.reactor.LongPollingMockConsumerSubject.assertThat;
+import static org.awaitility.Awaitility.await;
+
+@Slf4j
+class ReactorPCTest extends ParallelEoSStreamProcessorTestBase {
+
+    ReactorProcessor<String, String> rp;
+
+    @Override
+    protected ParallelEoSStreamProcessor initAsyncConsumer(ParallelConsumerOptions parallelConsumerOptions) {
+        var build = parallelConsumerOptions.toBuilder()
+                .commitMode(PERIODIC_CONSUMER_SYNC)
+                .maxConcurrency(10)
+                .build();
+
+        rp = new ReactorProcessor<>(build);
+
+        return rp;
+    }
+
+    @BeforeEach
+    public void setupData() {
+        super.primeFirstRecord();
+    }
+
+
+    @Test
+    void kickTires() {
+//        subscribeParallelConsumerAndMockConsumerTo(INPUT_TOPIC);
+        primeFirstRecord();
+        primeFirstRecord();
+        primeFirstRecord();
+
+        ConcurrentLinkedQueue<Object> msgs = new ConcurrentLinkedQueue<>();
+
+        rp.react((rec) -> {
+            log.info("Reactor user poll function: {}", rec);
+            msgs.add(rec);
+            Mono<String> result = Mono.just(StringUtils.msg("result: {}:{}", rec.offset(), rec.value()));
```

Comment:
```suggestion
            Mono<String> result = HttpClient.create()
                .post()
                .uri("https://httpbin.org/delay/2")
                .responseSingle((res, byteBufMono) -> {
                    log.info("Response code: " + res.status());
                    if (res.status() == HttpResponseStatus.OK) {
                        return byteBufMono.asString();
                    } else {
                        return Mono.error(new RuntimeException("http request failed with code " + res.status().code()));
                    }
                });
```

Reply:
Something like this? (please note the httpbin endpoint will respond with a 2s delay)

dependency: https://github.com/reactor/reactor-netty
```import reactor.netty.http.client.HttpClient;```

Reply:
please note for a real scenario one could even ignore the response body (byteBufMono) and simply check `res.status()` and e.g. `return Mono.just(res.status().code())` in case of success...

Reply:
Yeah that'd be cool. Chuck it in the example app.